packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1521,6 +1566,61 @@ export class TransactionController extends BaseController<
   ): boolean {
     return remoteGasUsed !== localGasUsed;
   }
+
+  private requestApproval(txMeta: TransactionMeta) {
+    const id = this.getApprovalId(txMeta);
+    const { origin } = txMeta;
+    const type = ApprovalType.Transaction;
+
+    this.messagingSystem
+      .call(
+        'ApprovalController:addRequest',
+        {
+          id,
+          origin: origin || ORIGIN_METAMASK,
+          type,
+          requestData: {
+            id: txMeta.id,
+            origin: origin || ORIGIN_METAMASK,
+            transaction: {
+              ...txMeta.transaction,
+            },
+          },
+        },
+        true,
+      )
+      .catch(() => {
```

Comment:
The problem in the extension was we only caught a rejected promise, rather than traditional errors which can still be thrown here as we're not using `await`.

Safest to `await` the `call` method, and wrap it in a `try catch`, we'll also need to make this method `async` but we don't need to `await` it when used.

Reply:
Did we change anything for this?

We want to do a `try catch` around the call and `await` it to ensure we catch all errors.

Reply:
Sorry my bad, I misunderstood. Wrapped messagingSystem call into a try catch and awaited.